control/src/lib.rs

```diff
@@ -356,82 +292,120 @@ pub mod pallet {
+				index, creator: sender.clone(), prime: sender.clone(), name, cid, org_type,
+				fee_model: fee_model.clone(), membership_fee: membership_fee.clone(), gov_currency,
+				pay_currency, member_limit, access_model, created: now.clone(), mutated: now,
+			};
+			let org_id = T::Hashing::hash_of(&org);
 			ensure!(!Orgs::<T>::contains_key(&org_id), Error::<T>::OrganizationExists);
 
-			Self::do_create_org(
-				sender.clone(), org_id.clone(), controller_id.clone(), treasury_account_id.clone(), name, cid,
-				org_type, access, fee_model.clone(), fee.clone(), gov_asset, pay_asset, member_limit, org_deposit, nonce
-			)?;
-			Self::do_add_member(org_id.clone(), controller_id, fee, fee_model, ControlMemberState::Active)?;
-			Self::mint_nft()?;
-
-			Self::deposit_event(Event::OrgCreated {
-				sender_id: sender,
-				org_id,
-				treasury_id: treasury_account_id,
-				created_at: <frame_system::Pallet<T>>::block_number(),
-				// TODO: tangram::NextRealmIndex::get()
-				realm_index: 0,
+			Self::do_create_org(org_id, &org, treasury_id.clone(), deposit)?;
+			Self::do_add_member(org_id, sender.clone(), MemberState::Active)?;
+			Self::pay_membership_fee(&sender, &treasury_id, membership_fee, fee_model, gov_currency)?;
+
+			Ok(())
+		}
+
+		#[pallet::weight(T::WeightInfo::update_org())]
+		pub fn update_org(
+			origin: OriginFor<T>,
+			org_id: T::Hash,
+			prime_id: Option<T::AccountId>,
+			access_model: Option<AccessModel>,
+			member_limit: Option<MemberLimit>,
+			fee_model: Option<FeeModel>,
+			membership_fee: Option<T::Balance>,
+		) -> DispatchResult {
+			let mut org = Orgs::<T>::get(&org_id).ok_or(Error::<T>::OrganizationUnknown)?;
+			Self::ensure_root_or_prime(origin, org.prime.clone(), org.org_type.clone())?;
+
+			let args = [prime_id.is_some(), fee_model.is_some(), membership_fee.is_some(), access_model.is_some(), member_limit.is_some()];
+			ensure!(args.iter().any(|x| *x == true), Error::<T>::NoChangesProvided);
+
+			if let Some(access_model) = access_model.clone() { org.access_model = access_model; };
+			if let Some(member_limit) = member_limit.clone() { org.member_limit = member_limit; };
+			if let Some(_) = membership_fee.clone() { org.membership_fee = membership_fee; };
+			if let Some(fee_model) = fee_model.clone() {
+				if fee_model != FeeModel::NoFees && membership_fee.is_none() {
+					return Err(Error::<T>::MissingParameter)?
+				};
+				org.fee_model = fee_model;
+			};
+			if let Some(prime_id) = prime_id.clone() {
+				org.prime = prime_id;
+				OrgPrime::<T>::insert(&org_id, org.prime.clone());
+			};
+
+			Orgs::<T>::insert(&org_id, org);
+			
+			let block_number = frame_system::Pallet::<T>::block_number();
+			Self::deposit_event(Event::OrgUpdated {
+				org_id, prime_id, access_model, member_limit,
+				fee_model, membership_fee, block_number
 			});
+
 			Ok(())
 		}
 
 		/// Enable Org
 		///
 		/// Enables an Org to be used and changes it's state to Active.
-		/// Root origin only.
```

Comment:
Probably specify `Root or prime only.`?

Reply:
in that case the question is: who would disable e.g. a vote? why and how?
- root only exists now, later on there is not root
- a prime might have bad intensions, but could/should be able to lock an org if he wants to "have a break"
- usually this would be imposed by either a DAO internal council vote
- in case of bad actors, disabling could be imposed by a Protocol Council Vote, like a circuit breaker.

therefore it must be root, e.g. 3/4 majority council ( tbd as per last discussion @5-mark re majorities and quorums)

Reply:
@2075 I agree with these points.
However documentation should reflect current logic and not the one, which we would like to achieve in some future. 
At the moment, voting is not yet implemented here. Once it will, we should update documentation accordingly.

Reply:
The design idea, as I understand it, is next:
- Control pallet interface allows direct changes for root or prime if `OrgType::Individual`.
- Signal pallet interface allows indirect changes through governance mechanisms if `OrgType::Dao` - **TBD**.

@2075, does it make sense, or should we reconsider it?

And talking about the Control pallet scope (no governance involved), what's your suggestion here?
At the moment, the permissions model for Control pallet extrinsics is:
- `add_member` - root or prime
- `remove_member` - root or prime
- `enable_org` - root or prime
- `disable_org` - root or prime
- `spend_funds` - root or prime

@2075, what should be changed here?

Also, `ensure_root_or_prime` idea was partially discussed here https://github.com/gamedaoco/gamedao-protocol/issues/75 for the spend_funds extrinsic, I've just extrapolated it for everything else.

cc: @5-mark